### ISSUE TITLE
Citrinet 256 and 512 for WSJ - limited resources

### ISSUE THREAD

**Issue by Miamoto** [NVIDIA/NeMo#3324]
Hello!  I wanna get baselines for citrinet-256 and citrinet-512 using WSJ, so that I can perform some experiments after. I will use speed perturbation and specAugment.

For now, I am training citrinet-256  with 4 GPUs, and batch 60. I also created a tokenizer with 250 for vocab and using WPE (not sure if this is the best for low resources).  I also decided to use dropout equal to 0.2 since there was a huge gap between validation and train in the first epochs. It is also relevant to note that I moved from learning rate of 0.01 to 0.001 since the loss was going to NaN after epoch 8. Also, as a note, I experimented to set the flag detect anomaly to True in the Trainer and it got an error in the first steps for getting NaN results in the backward pass - not sure if that is okay or not to happen...

Below are some configurations that I am using.
<img width="1010" alt="Captura de ecrã 2021-12-13, às 20 36 52" src="https://user-images.githubusercontent.com/15928244/145884925-dcf112e1-d427-4df4-8931-564001d534e0.png">
<img width="1015" alt="Captura de ecrã 2021-12-13, às 20 37 23" src="https://user-images.githubusercontent.com/15928244/145884979-f9a57682-d06c-493e-b43c-91b7ae6e5f6b.png">

It would be great to know if I am in the right direction! Thanks a lot

**Comment by titu1994** [comment]
If you don't need to train from scratch, I'd suggest vocab sizes of 1024 and load up both encoder and decoder checkpoint weights into the new model and fine time on WSJ. Use SPE Unigram tokenizer. 

However for scratch training, novograd LR = 0.01, betas = (0.8,0.25). 0.001 LR won't train the model at all. I don't know why nans are occuring, but if you're trying fp16 id suggest reducing batch size and using fp32 instead. 

Don't use dropout for small models, they don't have much capacity to work with.

**Comment by Miamoto** [comment]
Yea, I am using fp16. Do I reduce batch for what, 32? 

Thanks

**Comment by titu1994** [comment]
Reduced LR should normally deal with nans but if it occurs even at 0.01, then I would suggest changing to fp32 and train with batch size of 32
